### PR TITLE
Pin tinyxml2 to 8 + add migrator to 9

### PR DESCRIPTION
**Pull request by Tobias-Fischer**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by Tobias-Fischer** [comment]
tinyxml2 is used in quite a few feedstocks, and the recent bump from version 8 to 9 led to some issues where packages needed to be manually re-build in order to pull in the newer version of tinyxml2.
It would be great if this could be done with migrations in the future.


/cc @traversaro @wolfv

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -289,6 +289,8 @@ pin_run_as_build:
     max_pin: x.x.x
   sqlite:
     max_pin: x
+  tinyxml2:
+    max_pin: x.x
```
Comment (xhochy) [inline recipe/conda_build_config.yaml:293]: `tinyxml2` has a `run_exports` https://github.com/conda-forge/tinyxml2-feedstock/blob/8895462aa37083e7c32d676252887b7b7baba1f0/recipe/meta.yaml#L17 so this is not needed.
```suggestion
```

File: recipe/conda_build_config.yaml [hunk 2]
```diff
@@ -669,6 +671,8 @@ tbb_devel:
   - 2020
 thrift_cpp:
   - '0.14.1'
+tinyxml2:
+  - 9.0
```
Comment (xhochy) [inline recipe/conda_build_config.yaml:675]: Should we add `8` here and also add a migration at the same time or are you sure that everything is already migrated?

**Comment by isuruf** [comment]
Looking at https://abi-laboratory.pro/?view=timeline&l=tinyxml2, the pinning should be `x` instead of `x.x`.

**Comment by Tobias-Fischer** [comment]
Fix for the pinning in the tinyxml2 feedstock: https://github.com/conda-forge/tinyxml2-feedstock/pull/10

File: recipe/conda_build_config.yaml (hunk 2 above)
Comment (Tobias-Fischer) [reply recipe/conda_build_config.yaml]: Agreed - have done both. There are quite a few packages left that need migration.

**Comment by Tobias-Fischer** [comment]
Okay - I think I have addressed all of your comments @xhochy @isuruf - many thanks for the review.

**Pull request merged**